### PR TITLE
allow a string as selection on aupac-ember-data-typeahead

### PR DESCRIPTION
**Pull request by ssendev**
as far as i can tell the selection is never used to render templates so this shouldn't cause any errors

### CODE REVIEW

**Comment by jackmatt2** [comment]
@ssendev how could the returned `ember-data` model ever be a string?  Can you give a quick example?

**Comment by jackmatt2** [comment]
If you are thinking of passing a string as the initial selection, that will cause issues say if you have validations or something on the control as the typeahead will be showing a value but the `_selection` will remain invalid.

**Comment by jackmatt2** [comment]
Can I suggest a computed property whereby the actual model is fetched using the string.

``` javascript
initialValue : computed(function() {
    const something = this.get('model.something');
    return this.store.findRecord('your-model', { something : something});
});
```

**Comment by ssendev** [comment]
yes it is for use as initial selection. but validations are outside the typeahead so can be dealth with there. my usecase is that i store the important part as a query param and the other stuff (including the id) is not of concern so it dosn't matter that the _selection is not a full model. 

in fact in this case the query is a group by with average so there is no simple id i could be using. i had created a computed property which looked something like this

``` js
initialValue: computed(function(){
    const something = this.get('someQueryParam);
    this.store.createRecord('your-model', { something : something, id : 'typeahead'});
});
```

but this didn't work as it is the id property that is observed so it would not update on klicking the back button in the browser. my second try was to use `guidFor({})` to create a new id every time but this would create infinite records and would require cleanup of some sort. now that i'm writing this i had the idea to use `id : 'typeahead:' + something` which would limit the records created but is still not very nice.

concerning validations at least in the demo 'task 2' is written in the input but selected task is empty (which causes validation to fail if one focuses the typeahead and unfocuses again without entering text). which led me to believe it is on purpose and selected task had to be set manually. especialy considering the deprecation warnings on setting stuff while rendering, which this would require (maybe the warning can be avoided with using an acition but it would still produce a recursive dependency)

**Comment by jackmatt2** [comment]
mmmm, I still think this use case is kind of specific to your scenario - I might merge I just need to think about the impact a little more.  

I don't want this to hold you up so can I suggest you extend the `aupac-ember-data-typeahead` component and override the setValue method in your own project?

```
//yourapp/app/components/ssendev-ember-data-typeahead
import AupacEmberDataTypeahead from `ember-aupac-typeahead/components/aupac-ember-data-typeahead`

export default AupacEmberDataTypeahead.extend({

   setValue : function(selection) {
      //your override here
   }

});
```

I will take a look at those `deprecation` warnings you mentioned also.

**Comment by ssendev** [comment]
currently there are no deprecation warnings triggered and there wouldn't be any since selection would be set in an action. what i meant is that in order to avoid validation warnings it would be required to call the passed action with the passed selection immediately on instantiation. 

that way the outside would be informed that something is already selected but this would / could cause a dependency cycle (initial selection gets passed in -> action is called -> action sets value (e.g. the queryParam) -> selection is updated -> action is called ...) it would probably not run forever since compted propertys only trigger observers  if the value changed.

to avoid this one would make sure the validated value is already set and then only sets the selection.

in the demo for example `task` is `null` on the application controller even though the typeahead displays `task 2`. the used validations only trigger on blur so it is easy to miss but if one were to "submit" the form task would be `null` even though it looks as if `task 2` is selected. if the action would be called on init this would'nt be the case. the alternative is to set task immediately e.g. with `setInitialTask: on('init', function(){ this.set('task', this.get('initialtask')) })` on ApplicationController.

tl;dr &nbsp; the way things are displayed and internal value are not synchronized even when using an ember-data model as initial value, so using a string makes no difference

**Comment by jackmatt2** [comment]
@ssendev :+1: I stand corrected.  The initial value has nothing to do with the validation state in standing with Ember 2.0 data down, actions up approach.  I was getting confused the the old 2 way binding.  I will get this merged.

**Comment by jackmatt2** [comment]
Updated as of 0.0.10
